schema_based_on_diagram.sql

```diff
@@ -0,0 +1,57 @@
+
+CREATE DATABASE clinicdatabase;
+\c clinicdatabase;
+CREATE TABLE patients(
+    id INT NOT NULL PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    name VARCHAR(200),
+    date_of_birth DATE
+);
+
+CREATE TABLE medical_histories(
+    id INT NOT NULL PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    admitted_at TIMESTAMP,
+    patient_id INT,
+    status VARCHAR(200),
+    CONSTRAINT patient_fkey FOREIGN KEY (patient_id) REFERENCES patients(id)
+);
+
+CREATE TABLE invoices(
+    id INT NOT NULL PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    total_amount DECIMAL,
+    generated_at TIMESTAMP,
+    payed_at TIMESTAMP,
+    medical_history_id INT,
+    CONSTRAINT medical_history_fkey FOREIGN KEY(medical_history_id) REFERENCES medical_histories(id) 
+);
+
+CREATE TABLE treatments(
+    id INT NOT NULL PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    type VARCHAR(255),
+    name VARCHAR(255)
+);
+
+CREATE TABLE invoice_items(
+    id INT NOT NULL PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    unit_price DECIMAL,
+    quantity INT,
+    total_price DECIMAL,
+    invoice_id INT,
+    treatment_id INT,
+    CONSTRAINT invoice_id_fkey FOREIGN KEY(invoice_id) REFERENCES invoices(id),
+    CONSTRAINT treatments_id_fkey FOREIGN KEY(treatment_id) REFERENCES treatments(id)
+);
+
+CREATE TABLE medical_records(
+    id INT NOT NULL PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    medical_histories_id INT,
+    treatment_id INT,
+    CONSTRAINT medical_records_fkey FOREIGN KEY(medical_histories_id) REFERENCES medical_histories(id),
+    CONSTRAINT treatments_fkey FOREIGN KEY(treatment_id) REFERENCES treatments(id) 
+);
+
+
+
+CREATE INDEX medical_patient_index ON medical_histories(patient_id);
+CREATE INDEX medical_history_fkey_index ON invoices(medical_history_id);
+CREATE INDEX medical_records_fkey_index ON medical_records(medical_histories_id);
+CREATE INDEX treatments_fkey_index ON medical_records(treatment_id);
```

Comment:
- [ ] @Team great work adding indexes on foreign keys, however, you forgot to add keys for invoice_id in invoice_items and treatment_id in invoice_items, kindly consider adding indexes to them.